### PR TITLE
src: emmc: Check and assign default offset to first partition

### PR DESCRIPTION
**Pull request by mschwan-phytec**
Check, whether the first partition has a valid offset being specified. If no offset was specified, use a sensible default. In all cases, protect the partition table from being potentially overwritten by the first partition.

### CODE REVIEW

**Comment by mschwan-phytec** [comment]
This PR requires #25 to be merged first.

File: src/emmc.c [hunk 1]
```diff
@@ -706,6 +708,29 @@ pu_emmc_parse_partitions(PuEmmc *emmc,
             return FALSE;
         }
 
+        if (first_partition) {
+            first_partition = FALSE;
+            PedSector min_offset = 1;
+
+            if (g_str_equal(emmc->disktype->name, "msdos"))
+                min_offset = PARTITION_TABLE_SIZE_MSDOS;
```
Comment (landerweit-phytec) [inline src/emmc.c:716]: Initialize `min_offset` with `PARTITION_TABLE_SIZE_MSDOS` as `msdos` is the default partition table.

**Review by landerweit-phytec — CHANGES_REQUESTED** [review]
Looks good. Only one thing below.

On another note, should we also worry about the output-offset of raw data?

**Comment by mschwan-phytec** [comment]
> On another note, should we also worry about the output-offset of raw data?

This is a valid point, but something we should tackle in a different PR, as much more checking is involved with that. I will create a ticket in our internal issue tracker.

**Review by landerweit-phytec — APPROVED** [review]

**Pull request merged**